### PR TITLE
[fast-double-parser] Add the Fast Double Parser port

### PR DESCRIPTION
**Pull request by dsmith111**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
fast-double-parser is header-only and can be used from CMake via:

  find_path(FAST_DOUBLE_PARSER_INCLUDE_DIRS "fast_double_parser.h")
  target_include_directories(main PRIVATE ${FAST_DOUBLE_PARSER_INCLUDE_DIRS})
```

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/fast-double-parser/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "fast-double-parser",
+  "version": "0.7.0",
```
Comment (MonicaLiu0311) [inline ports/fast-double-parser/vcpkg.json:3]: One question: 0.7.0 was released in 2022, why not just update to [0.8.0](https://github.com/lemire/fast_double_parser/releases/tag/v0.8.0), which is the lastest version.

Comment (dsmith111) [reply ports/fast-double-parser/vcpkg.json:3]: See the mentioned PR. I was asked to devendor the lightgbm port. The LightGBM package is pinned to this version of fast double parser. But, when the LightGBM library is updated to use the v0.8.0 fast double parser library. I'll of course have to update the version here.

Comment (dsmith111) [reply ports/fast-double-parser/vcpkg.json:3]: For ubiquity:

https://github.com/microsoft/vcpkg/pull/39395#discussion_r1648417216

https://github.com/microsoft/LightGBM/tree/master/external_libs

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**